### DISCUSSION TITLE
Does AccelSim's Trace-based execution execute host-side syscalls like file handling?

### DISCUSSION THREAD

**Discussion by dpapagiannis** [accel-sim/accel-sim-framework#197] [Q&A] [answered]
From my understanding, GPGPU-Sim is given an executable, and the host-side part of it is executed on the machine running the simulation, while the CUDA calls are intercepted by the simulator.

Since AccelSim is running a trace and not an executable, is the host-side part of the executable executed at all?

The question I'm actually trying to answer is whether a traced CUDA program that should be producing an output file on host will produce the file when running on AccelSim or not.

**Example:** The `hotspot` benchmark in `rodinia-2.0-ft` produces an `output_pyramid.dat` file, when run on a GPU (also on GPGPU-Sim). There _is_ such a file inside the pre-traced program's directory (in our example: `./hw_run/rodinia_2.0-ft/11.0/hotspot-rodinia-2.0-ft/30_6_40___data_result_30_6_40_txt`), but I assume that this file is created only once, when the trace is created by running the benchmark via the NVbit tool on an actual GPU. Is this correct?

This may be answered elsewhere and I may have missed it.

**Comment by JRPan** [comment, marked as answer]
No, there are no output files when running with trace model. And you are correct with the example.

**Reply by dpapagiannis** [reply]
Thanks a lot for the reply @JRPan !

I will dig a bit deeper and ask the following:

I am interested in performing fault injections in the memory hierarchy, so an output file to compare results would be very useful. Do you think there is any workaround to achieving that? E.g. some way to use the `cudaMemcpy` functionality from GPGPU-Sim? I have no idea how difficult that would be, though.
